dracut-saltboot/saltboot/saltboot.sh

```diff
@@ -224,16 +215,35 @@ if [ -z "$CUR_MASTER" -o "salt" == "$CUR_MASTER" ] ; then
     if [ -z "$MASTER" ] ; then
         MASTER=`dig $DIG_OPTIONS -t CNAME salt.$DNSDOMAIN | sed -e 's|;;.*||' -e 's|\.$||' `
     fi
-    if [ -n "$MASTER" ] ; then
-        echo "master: $MASTER" > $INITRD_SALT_ETC/minion.d/master.conf
-    fi
-    # else
-    # ... if it fails, do nothing and let it fall back to 'salt'
-    # by continuing we hide dns errors. Shouldn't we require proper setup?
 fi
 
 Echo "Using Salt master: ${MASTER:-$CUR_MASTER}"
 
+if ! grep -q "^master: ${MASTER:-$CUR_MASTER}$" $INITRD_SALT_ETC/minion.d/susemanager.conf ; then
```

Comment:
This will work on nonexisting susemanager.conf too, but will print file not found error. Can you add `2>/dev/null`?

```suggestion
if ! grep -q "^master: ${MASTER:-$CUR_MASTER}$" $INITRD_SALT_ETC/minion.d/susemanager.conf 2>/dev/null ; then
```